spec/factories.rb

```diff
@@ -257,10 +257,29 @@
 
     factory :degree_qualification do
       level { 'degree' }
-      qualification_type { Hesa::DegreeType.all.sample.name }
-      subject { Hesa::Subject.all.sample.name }
-      institution_name { Hesa::Institution.all.sample.name }
-      grade { Hesa::Grade.all.sample.description }
+
+      after(:build) do |qualification|
+        if qualification.qualification_type.blank?
+          hesa_type = Hesa::DegreeType.all.sample
+          qualification.qualification_type = hesa_type.name
+          qualification.qualification_type_hesa_code = hesa_type.hesa_code
+        end
+        if qualification.subject.blank?
+          hesa_subject = Hesa::Subject.all.sample
+          qualification.subject = hesa_subject.name
+          qualification.subject_hesa_code = hesa_subject.hesa_code
+        end
+        if qualification.institution_name.blank?
+          hesa_institution = Hesa::Institution.all.sample
+          qualification.institution_name = hesa_institution.name
+          qualification.institution_hesa_code = hesa_institution.hesa_code
+        end
+        if qualification.grade.blank?
+          hesa_grade = Hesa::Grade.all.sample
+          qualification.grade = hesa_grade.description
+          qualification.grade_hesa_code = hesa_grade.hesa_code
+        end
+      end
```

Comment:
We're currently doing something like this in master, introduced in one of my PRs at some point. A rebase ought to reconcile the two!

Reply:
:+1: I think yours is better because it handles the case where the string values and not the codes are specified by the caller.